src/languages/c.ts

```diff
@@ -28,78 +32,66 @@ export default {
 			'number':
 				/(?:\b0x(?:[\da-f]+(?:\.[\da-f]*)?|\.[\da-f]+)(?:p[+-]?\d+)?|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:e[+-]?\d+)?)[ful]{0,4}/i,
 			'operator': />>=?|<<=?|->|([-+&|:])\1|[?:~]|[-+*/%&|^!=<>]=?/,
-		});
-
-		insertBefore(c, 'string', {
-			'char': {
-				// https://en.cppreference.com/w/c/language/character_constant
-				pattern: /'(?:\\(?:\r\n|[\s\S])|[^'\\\r\n]){0,32}'/,
-				greedy: true,
-			},
-		});
-
-		insertBefore(c, 'string', {
-			'macro': {
-				// allow for multiline macro definitions
-				// spaces after the # character compile fine with gcc
-				pattern:
-					/(^[\t ]*)#\s*[a-z](?:[^\r\n\\/]|\/(?!\*)|\/\*(?:[^*]|\*(?!\/))*\*\/|\\(?:\r\n|[\s\S]))*/im,
-				lookbehind: true,
-				greedy: true,
-				alias: 'property',
-				inside: {
-					'string': [
-						{
-							// highlight the path of the include statement as a string
-							pattern: /^(#\s*include\s*)<[^>]+>/,
-							lookbehind: true,
-						},
-						c['string'] as GrammarToken,
-					],
-					'char': c['char'],
-					'comment': c['comment'],
-					'macro-name': [
-						{
-							pattern: /(^#\s*define\s+)\w+\b(?!\()/i,
+			$insertBefore: {
+				'string': {
+					'char': {
+						// https://en.cppreference.com/w/c/language/character_constant
+						pattern: /'(?:\\(?:\r\n|[\s\S])|[^'\\\r\n]){0,32}'/,
+						greedy: true,
+					},
+				},
+				'macro': {
+					// allow for multiline macro definitions
+					// spaces after the # character compile fine with gcc
+					pattern:
+						/(^[\t ]*)#\s*[a-z](?:[^\r\n\\/]|\/(?!\*)|\/\*(?:[^*]|\*(?!\/))*\*\/|\\(?:\r\n|[\s\S]))*/im,
+					lookbehind: true,
+					greedy: true,
+					alias: 'property',
+					inside: {
+						'string': [
+							{
+								// highlight the path of the include statement as a string
+								pattern: /^(#\s*include\s*)<[^>]+>/,
+								lookbehind: true,
+							},
+							c['string'] as GrammarToken,
```

Comment:
We should define `c` somehow. Previously, it was defined as `const c = extend(...)`.

Reply:
Or eliminate the need to refer to it. 🤔